src/boot/efi/boot.c

```diff
@@ -1814,6 +1814,56 @@ static VOID config_entry_add_osx(Config *config) {
         }
 }
 
+
+static VOID config_entry_add_windows(Config *config, EFI_HANDLE *device, EFI_FILE *root_dir) {
+        EFI_STATUS err;
+        UINTN len;
+        const CHAR16 *title = NULL;
+        _cleanup_freepool_ CHAR8 *bcd = NULL;
+
+        if (!config->auto_entries)
+                return;
+
+        /* Try to find a better title. */
+        err = file_read(root_dir, L"\\EFI\\Microsoft\\Boot\\BCD", 0, 100*1024, &bcd, &len);
+        if (!EFI_ERROR(err)) {
+                const CHAR16 prefix[] = L"Windows ";
+                const UINTN prefix_len = sizeof(prefix) - sizeof(CHAR16);
```

Comment:
maybe move the STRLEN() macro from src/basic/ to src/fundamental/, bevcause that's what you are reimplementing here

Reply:
Did this for now. But I think this needs #20346 for the non sd-boot build to finish.